### PR TITLE
Tweet releases

### PR DESCRIPTION
**Pull request by fao89**
[noissue]

### CODE REVIEW

File: templates/github/.ci/scripts/tweet.py.j2 [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import os
+import sys
+from tweepy import Client
+
+release_version = sys.argv[1]
+client = Client(
+    consumer_key=os.getenv("TWITTER_API_KEY"),
+    consumer_secret=os.getenv("TWITTER_API_KEY_SECRET"),
+    access_token=os.getenv("TWITTER_ACCESS_TOKEN"),
+    access_token_secret=os.getenv("TWITTER_ACCESS_TOKEN_SECRET"),
```
Comment (fao89) [inline templates/github/.ci/scripts/tweet.py.j2:10]: we need to set these keys on github

Comment (RCMariko) [reply templates/github/.ci/scripts/tweet.py.j2:10]: @bmbouter has the info we need. I can set up the keys and he may also be able to do that - but we'll have to wait til he is back.

**Comment by goosemania** [comment]
Thank for putting this together, I need to think more about it.

I think we likely want to post only for Y releases and when it makes sense. E.g. if it's a pulpcore release with some breaking changes for plugins, we probably do not want to tweet about it immediately but wait for plugins. And when plugins are released, I guess it's pretty annoying to get N tweets fro each plugin, rather then one consolidated one for pulpcore and compatible plugins :thinking: 

I like the idea of posting a tweet easily though. Maybe a manual job which asks for a message to provide :thinking: will do the trick.

**Comment by bmbouter** [comment]
+1 to tweeting for Y-streams only. I haven't looked at the code to see what it actually does. I don't know how to make it fancier to wait for pulpcore+plugin releases, so I'd settle for just any Y release. I think we should auto-tweet something because with fewer people we need more automation.

File: templates/github/.ci/scripts/tweet.py.j2 [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import os
+import sys
+from tweepy import Client
+
+release_version = sys.argv[1]
+if release_version.endswith(".0"):
```
Comment (fao89) [inline templates/github/.ci/scripts/tweet.py.j2:6]: I already modified the code to be Y-stream only

**Comment by ipanova** [comment]
> Thanks for putting this together, I need to think more about it.
> 
> I think we likely want to post only for Y releases and when it makes sense. E.g. if it's a pulpcore release with some breaking changes for plugins, we probably do not want to tweet about it immediately but wait for plugins. And when plugins are released, I guess it's pretty annoying to get N tweets fro each plugin, rather then one consolidated one for pulpcore and compatible plugins thinking
We have every 5th release a breaking one so that should not be that big of the problem, I think.
> 
> I like the idea of posting a tweet easily though. Maybe a manual job which asks for a message to provide thinking will do the trick.

This is almost like going and doing the tweet yourself :shrug:

**Comment by ipanova** [comment]
> +1 to tweeting for Y-streams only. I haven't looked at the code to see what it actually does. I don't know how to make it fancier to wait for pulpcore+plugin releases, so I'd settle for just any Y release. I think we should auto-tweet something because with fewer people we need more automation.

+1

**Comment by goosemania** [comment]
> 
> > I like the idea of posting a tweet easily though. Maybe a manual job which asks for a message to provide thinking will do the trick.
> 
> This is almost like going and doing the tweet yourself shrug

Yup, and it's fine since folks do not need credentials or go anywhere else apart from github. IMO, reduces friction.

File: templates/github/.ci/scripts/tweet.py.j2 [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    client = Client(
+        consumer_key=os.getenv("TWITTER_API_KEY"),
+        consumer_secret=os.getenv("TWITTER_API_KEY_SECRET"),
+        access_token=os.getenv("TWITTER_ACCESS_TOKEN"),
+        access_token_secret=os.getenv("TWITTER_ACCESS_TOKEN_SECRET"),
+    )
+    link = "https://docs.pulpproject.org/{{ plugin_snake }}/changes.html"
+    msg = f"{{ plugin_name }}-{release_version} - Check it out at: {link}"
```
Comment (ipanova) [inline templates/github/.ci/scripts/tweet.py.j2:14]: "Check out for more details :...."

File: templates/github/.ci/scripts/tweet.py.j2 [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+release_version = sys.argv[1]
+if release_version.endswith(".0"):
+    client = Client(
+        consumer_key=os.getenv("TWITTER_API_KEY"),
+        consumer_secret=os.getenv("TWITTER_API_KEY_SECRET"),
+        access_token=os.getenv("TWITTER_ACCESS_TOKEN"),
+        access_token_secret=os.getenv("TWITTER_ACCESS_TOKEN_SECRET"),
+    )
+    link = "https://docs.pulpproject.org/{{ plugin_snake }}/changes.html"
```
Comment (ipanova) [inline templates/github/.ci/scripts/tweet.py.j2:13]: did you mean plugin_name?

Comment (fao89) [reply templates/github/.ci/scripts/tweet.py.j2:13]: nope, plugin_snake for ensuring the _
https://docs.pulpproject.org/pulp_2to3_migration/

**Review by ipanova — COMMENTED** [review]
looks good to me!

**Review by ipanova — APPROVED** [review]

File: templates/github/.ci/scripts/tweet.py.j2 [hunk 5]
```diff
@@ -0,0 +1,16 @@
+import os
+import sys
+from tweepy import Client
```
Comment (bmbouter) [inline templates/github/.ci/scripts/tweet.py.j2:3]: How is this dependency being installed?

Comment (bmbouter) [reply templates/github/.ci/scripts/tweet.py.j2:3]: Oh I see it further down. Great!

**Comment by fao89** [comment]
> @fao89 thank you for this. Please merge when you feel all comments are addressed (if not already). Please let me know if I need to configure any credentials anywhere for this to work.

we need to set these secrets on pulp org:
- TWITTER_API_KEY
- TWITTER_API_KEY_SECRET
- TWITTER_ACCESS_TOKEN
- TWITTER_ACCESS_TOKEN_SECRET

**Pull request merged**